packages/grafana-eslint-rules/rules/no-unreduced-motion.cjs

```diff
@@ -0,0 +1,62 @@
+// @ts-check
+const { ESLintUtils, AST_NODE_TYPES } = require('@typescript-eslint/utils');
+
+const createRule = ESLintUtils.RuleCreator((name) => `https://github.com/grafana/grafana/blob/main/packages/grafana-eslint-rules/README.md#${name}`);
+
+const restrictedProperties = ['animation', 'transition'];
+
+const rule = createRule({
+  create(context) {
+    return {
+      CallExpression(node) {
+        if (
+          node.callee.type === AST_NODE_TYPES.Identifier &&
+          node.callee.name === 'css'
+        ) {
+          const cssObjects = node.arguments.flatMap((node) => {
+            switch (node.type) {
+              case AST_NODE_TYPES.ObjectExpression:
+                return [node];
+              case AST_NODE_TYPES.ArrayExpression:
+                return node.elements.filter(v => v.type === AST_NODE_TYPES.ObjectExpression);
+              default:
+                return [];
+            }
+          });
+
+          for (const cssObject of cssObjects) {
+            if (cssObject.type === AST_NODE_TYPES.ObjectExpression) {
+              for (const property of cssObject.properties) {
+                if (
+                  property.type === AST_NODE_TYPES.Property &&
+                  property.key.type === AST_NODE_TYPES.Identifier &&
+                  restrictedProperties.some(prop => property.key.name.startsWith(prop))
+                ) {
+                  context.report({
+                    node: property,
+                    messageId: 'noUnreducedMotion',
+                  });
```

Comment:
Can you explain roughly what this is doing? My guess is:

 - Find all CSS objects like `css({...})` or `css([{...}, {...}])`
 - Flag any properties in them that start with animation or transition
 - and `handleReducedMotion` or a direct `prefers-reduced-motion` media query avoids getting flagged because those objects are not being inspected

Which means that

```ts
const ident => (obj) => obj;

css({
    ...ident({
        transition: '...',
    })
})

// or

css({
    'screen and (min-width)': {
        'transition': '...'
    }
})
```

could be used to avoid getting flagged by this lint rule?

I don't mean this as a criticism, just trying to understand exactly how this works.

Reply:
yep pretty much exactly 👍 

there's definitely gaps in the logic, but i think this would probably catch ~95%?

Reply:
That's fine.

The thing that's just confusing when reading this is that the rule is to enforce `handleReducedMotion` or `prefers-reduced-motion`, but thats never even checked in the rule's implementation. Just makes for a bit of a confusing read.

Maybe a tweaked implementation could be to more comprehensively gather css objects as `css()` arguments to inspect, and then could specifically allow through certain patterns.